pkg/lib/utils.tsx

```diff
@@ -21,23 +21,33 @@ import React from "react";
 
 import cockpit from "cockpit";
 
-export function fmt_to_fragments(fmt) {
-    const args = Array.prototype.slice.call(arguments, 1);
-
-    function replace(part) {
+export function fmt_to_fragments(format: string, ...args: React.ReactNode[]) {
```

Comment:
Does a `string` pass for a `ReactNode`?

Reply:
It does.  That's the thing that lets you say `<p>Hi!</p>`, or more explicitly:

```tsx
const content = "Hi!";
return <p>{content}</p>;
```

In that case, the `content` is passed as the `children` when constructing the `<p>` element.  `undefined`, `null` and `false` are also valid which covers the common idioms like `<>{ condition && content }</>` which evaluates to `false` when the content ought not to be included.

Reply:
I just went to look up the definition.  Good for a chuckle :)

```ts
    type ReactNode =
        | ReactElement
        | string
        | number
        | Iterable<ReactNode>
        | ReactPortal
        | boolean
        | null
        | undefined
        | DO_NOT_USE_OR_YOU_WILL_BE_FIRED_EXPERIMENTAL_REACT_NODES[
            keyof DO_NOT_USE_OR_YOU_WILL_BE_FIRED_EXPERIMENTAL_REACT_NODES
        ];

```